src/main/kotlin/org/jetbrains/research/libsl/visitors/FunctionVisitor.kt

```diff
@@ -94,16 +131,18 @@ class FunctionVisitor(
             expressionVisitor.visitExpression(expr)
         }.orEmpty().toMutableList()
 
+        context.storeAnnotation(Annotation(name, args))
+
         if (name == "target") {
-            return TargetAnnotation()
+            return AnnotationReference("target", context)
         }
-
-        return Annotation(name, args)
+        return AnnotationReference(name, context)
     }
 
     private val List<FunctionArgument>.getFunctionTargetByAnnotation: AutomatonReference?
         get() {
-            val targetArg = firstOrNull { arg -> arg.annotation?.name == "target" } ?: return null
+            val targetArg = firstOrNull { arg -> arg.annotationsReferences?.any {
+                it.resolveOrError().name == "target"} ?: false } ?: return null
```

Comment:
If you move out `@Target` annotation (see previous comment), this check cold be replaced with something like `it.resolveOrError() == TargetAnnotationHelper.targetAnnotation`